src/Polykey.ts

```diff
@@ -77,7 +79,17 @@ export default class Polykey {
     let vault: Vault
     try {
       vault = await this.getVault(vaultName)
-      vault.addSecret(secretName, secret)
+      await vault.addSecret(secretName, secret)
+    } catch(err) {
+      throw err
```

Comment:
Why catch just to throw?

Reply:
This has been taken out :+1: and the whole method moved to the `Vault` class